serializers/src/glTF/2.0/glTFMaterialExporter.ts

```diff
@@ -515,7 +515,15 @@ export class _GLTFMaterialExporter {
                 tempTexture.dispose();
 
                 // Read data from WebGL
-                const canvas = engine.getRenderingCanvas();
+                const canvas0 = engine.getRenderingCanvas();
+
+                let canvas: Nullable<HTMLCanvasElement> = document.createElement("canvas");
+
+                canvas.width = canvas0?.width ?? 0;
+                canvas.height = canvas0?.height ?? 0;
+
+                var destCtx = canvas.getContext('2d');
+                destCtx!.drawImage(canvas0!, 0, 0);

```

Comment:
I think the block of code above that copies the canvas is needed, else it's possible we render a picture whereas the previous one is still not processed by the code below (which converts the canvas to a blob then to a base64 string), overwritting the previous data in the canvas.

Or are we sure all the promises involved in the process will be completed right away, so in effect all the conversion code is synchronous (in which case no need to copy the canvas)? In effect, it's only `fileReader.readAsDataURL` that will trigger `fileReader.onLoad` that is/can be asynchronous. If we are sure it won't be done asynchronously, I can remove my code.

I tried to have a look at the [MDN](https://developer.mozilla.org/en-US/docs/Web/API/FileReader/readAsDataURL) and the [spec](https://w3c.github.io/FileAPI/#readAsDataURL), but I can't see anything saying that `readAsDataURL` would be synchronous...

Reply:
I think this is fine to be secure